code/Game.Moderation.cs

```diff
@@ -0,0 +1,104 @@
+using Sandbox;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace TTT;
+
+public struct BannedClient
+{
+	public long SteamId { get; set; }
+	public string Reason { get; set; }
+}
+
+public partial class Game : Sandbox.Game
+{
+	private const string BanFilePath = "bans.json";
+	private static List<BannedClient> _bannedClients = new();
+
+	public override bool ShouldConnect( long playerId )
+	{
+		return !_bannedClients.Any( ( bannedClient ) => bannedClient.SteamId == playerId );
+	}
+
+	private void LoadBannedClients()
+	{
+		var clients = FileSystem.Data.ReadJson<List<BannedClient>>( BanFilePath );
+		if ( !clients.IsNullOrEmpty() )
+			_bannedClients = clients;
+	}
+
+	[ConCmd.Admin( Name = "ttt_ban_name", Help = "Ban the user with the name" )]
+	public static void BanPlayerWithName( string name, string reason = "" )
+	{
+		foreach ( var client in Client.All )
+		{
+			if ( client.Name != name )
+				continue;
+
+			BanPlayer( client, reason, true );
+			break;
+		}
+	}
+
+	[ConCmd.Admin( Name = "ttt_ban_steamid", Help = "Ban the user with the steamid" )]
+	public static void BanPlayerWithSteamID( long steamId, string reason = "" )
+	{
+		foreach ( var client in Client.All )
```

Comment:
I should give feedback on all these commands on whether or not the player was found.